### PR TITLE
Highlander Jobs / Better Job Cache

### PR DESCRIPTION
**Pull request by darkside**
Don't run a job with the same signature more than once per request. Unless we want to :)

### CODE REVIEW

File: lib/coney_island/jobs_cache.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      self
+    end
+
+    def cache_job(*args)
+      self.cached_jobs[generate_id(*args)] = args
+    end
+
+    def flush_jobs
+      while job = self.cached_jobs.shift
+        publish_job(job[1], job[0])
```
Comment (darkside) [inline lib/coney_island/jobs_cache.rb]: @edraut I think this is a more intuitive behaviour for flushing, it clears out the Hash as it publishes them.

File: lib/coney_island/jobs_cache.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    def caching_jobs
+      @adapter.store[:caching_jobs]
+    end
+
+    def caching_jobs=(boolean)
+      @adapter.store[:caching_jobs] = boolean
+    end
+
+    def generate_id(*args)
```
Comment (darkside) [inline lib/coney_island/jobs_cache.rb]: This is where the actual magic happens. We may have to test / deal with stringified keys.

**Comment by edraut** [comment]
@darkside this looks great. I just realized we don't have a BugSnag adapter for Coney, could you add that before you release this version? It should be pretty quick, let me know if you have questions.

File: test/jobs_cache_test.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class JobsCacheTest < MiniTest::Test
   describe ConeyIsland::JobsCache do
     before do
       @instance = ConeyIsland::JobsCache.new
+      @instance.flush_jobs
```
Comment (darkside) [inline test/jobs_cache_test.rb]: God damn it

File: lib/coney_island/jobs_cache.rb (hunk 2 above)
Comment (gomayonqui) [inline lib/coney_island/jobs_cache.rb]: :+1:

**Comment by gomayonqui** [comment]
:+1: great job

**Comment by edraut** [comment]
great work @darkside ! 👍

**Pull request merged**